core-bundle/src/EventListener/UnwrapTwigExceptionListener.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\EventListener;
+
+use Contao\CoreBundle\Exception\NoContentResponseException;
+use Contao\CoreBundle\Exception\RedirectResponseException;
+use Symfony\Component\HttpKernel\Event\ExceptionEvent;
+use Twig\Error\RuntimeError;
+
+class UnwrapTwigExceptionListener
+{
+    private static array $exceptionsToUnwrap = [
+        NoContentResponseException::class,
+        RedirectResponseException::class,
+    ];
+
+    /**
+     * If an exception is encountered while rendering a Twig template, Twig
+     * will wrap the exception in a Twig\Error\RuntimeError. For cases that we
+     * want the exception to bubble, we need to unwrap the original exception
+     * again.
+     */
+    public function __invoke(ExceptionEvent $event): void
+    {
+        if (
+            !($throwable = $event->getThrowable()) instanceof RuntimeError ||
+            ($previous = $throwable->getPrevious()) === null ||
+            !\in_array(\get_class($previous), self::$exceptionsToUnwrap, true)
```

Comment:
Should we iterate over the array and use a `$previous instanceof $exceptionToUnwrap` check instead? This way we would also detect child classes which is the expected behaviour I think.

Reply:
Changed in 0ccd80afb08f4f239919a56b9681eb93582631b5.